### PR TITLE
Add support for custom labels on Cluster CR

### PR DESCRIPTION
**Pull request by primeroz**
Towards https://github.com/giantswarm/giantswarm/issues/24718

### Please check if PR meets these requirements

- [ ] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Comment by primeroz** [comment]
@marians if you could have a look 

the validation is failing with

```
---

Errors (1)

Object must have at least one property
  - /metadata/clusterLabels
```

but what i actually want is , similar to https://github.com/giantswarm/cluster-cloud-director/pull/71/files , an empty object by default 

how do i get that with our current schema rules ? 

thanks

**Comment by marians** [comment]
In https://github.com/giantswarm/cluster-cloud-director/pull/161 I have done something similar. Ideally we would align this as much as possible. (The main difference is that in cluster-cloud-director the labels are also applied to the HelmRelease resource.)

Please call the property `.metadata.labels`. Since the entire config is about a cluster, the word "cluster" is not needed here.

Also I would favour using `patternProperties` here like in https://github.com/giantswarm/cluster-cloud-director/pull/161/files#diff-e2aa539a2b05339dcc4afe7112e00ef491c55342eef3f5e79d21ba5d12e8be7aR629-R637 to prevent user input that does not qualify as valid K8s label key (e. g. string containing a space).

**Comment by primeroz** [comment]
@marians all your suggestions are applied now, please have a look :pray:

File: helm/cluster-azure/values.schema.json [hunk 1]
```diff
@@ -352,14 +352,19 @@
...
-                    "title": "Cluster Labels",
-                    "description": "Map of labels to be added to the cluster CR.",
-                    "additionalProperties": {
-                        "type": "string"
-                    },
-                    "default": {}
+                    "title": "Labels",
+                    "description": "These labels are added to the Kubernetes resourses defining this cluster.",
+                    "patternProperties": {
+                        "^[_-a-zA-Z0-9\\.]$": {
```
Comment (marians) [inline helm/cluster-azure/values.schema.json:360]: Ran my own regex against https://www.regexpal.com/ and found some mistakes. This should fix them.

```suggestion
                        "^[a-zA-Z0-9\\._-]+$": {
```

File: helm/cluster-azure/values.schema.json [hunk 2]
```diff
@@ -352,14 +352,19 @@
...
+                            "type": "string",
+                            "title": "Label",
+                            "maxLength": 63,
+                            "minLength": 0,
+                            "pattern": "^[_-a-zA-Z0-9\\.]$"
```
Comment (marians) [inline helm/cluster-azure/values.schema.json:365]:
```suggestion
                            "pattern": "^[a-zA-Z0-9\\._-]+$"
```

**Review by njuettner — APPROVED** [review]

**Pull request merged**